packages/fxa-settings/src/components/NotificationPromoBanner/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { Link, RouteComponentProps, useLocation } from '@reach/router';
+
+type NotificationPromoBannerProps = {
+  headerImage: string;
+  ctaText: string;
+  headerValue: string;
+  headerDescription: string;
+  route: string;
+  dismissKey: string;
+  isVisible: boolean;
+};
+
+const NotificationPromoBanner = ({
+  headerImage,
+  ctaText,
+  headerValue,
+  headerDescription,
+  dismissKey,
+  isVisible,
+  route,
+}: NotificationPromoBannerProps & RouteComponentProps) => {
+  const location = useLocation();
+
+  // The visibility state of the banner should be determined by the parent component
+  // since it knows if the user has already enabled the feature being promoted.
+  const [visible, setVisible] = useState(isVisible);
+
+  // This is value is used to animate the banner out of view when the user dismisses it.
+  const [isClosing, setIsClosing] = useState(false);
+
+  const notificationBannerClosedLocalStorageKey = `__fxa_storage.fxa_disable_notification_banner.${dismissKey}`;
+
+  const [bannerClosed] = useState<string | null>(
+    localStorage.getItem(notificationBannerClosedLocalStorageKey)
+  );
+
+  if (!visible || bannerClosed === 'true') return null;
+
+  return (
+    <div
+      className={`relative flex justify-between items-center rounded-lg bg-gradient-to-tr from-blue-600/10 to-purple-500/10 p-3 shadow-md space-x-4 mb-5 transition-transform duration-300 ease-in-out ${
+        isClosing ? 'translate-x-full' : 'translate-x-0'
+      }`}
+    >
+      <button
+        id={`close-${dismissKey}`}
+        data-testid={`close-${dismissKey}`}
+        onClick={() => {
+          localStorage.setItem(notificationBannerClosedLocalStorageKey, 'true');
+          setIsClosing(true);
+          setTimeout(() => setVisible(false), 300);
+        }}
+        className="absolute top-0 right-0 mr-4 text-lg"
```

Comment:
For the follow-up, we should note to test for RTL mode - for example we can use the tailwind logical properties `end-`, `me-4` here.